Modules/Loadable/Segmentations/EditorEffects/Python/SegmentEditorIslandsEffect.py

```diff
@@ -149,57 +149,77 @@ def splitSegments(self, minimumSize = 0, maxNumberOfSegments = 0, split = True):
-        labelValue = int(labelValues.GetTuple1(i))
-        if i == 0:
-          segment = selectedSegment
-          segmentID = selectedSegmentID
-        else:
-          segment = slicer.vtkSegment()
-          name = baseSegmentName + "_" + str(i+1)
-          segment.SetName(name)
-          segment.AddRepresentation(slicer.vtkSegmentationConverter.GetSegmentationBinaryLabelmapRepresentationName(),
-            selectedSegment.GetRepresentation(slicer.vtkSegmentationConverter.GetSegmentationBinaryLabelmapRepresentationName()));
-          segmentation.AddSegment(segment)
-          segmentID = segmentation.GetSegmentIdBySegment(segment)
-          segment.SetLabelValue(segmentation.GetUniqueLabelValueForSharedLabelmap(selectedSegmentID))
-
-        threshold = vtk.vtkImageThreshold()
-        threshold.SetInputData(islandMath.GetOutput())
-        threshold.ThresholdBetween(labelValue, labelValue)
-        threshold.SetInValue(1)
-        threshold.SetOutValue(0)
-        threshold.Update()
-
-        # Create oriented image data from output
-        modifierImage = slicer.vtkOrientedImageData()
-        modifierImage.DeepCopy(threshold.GetOutput())
-        selectedSegmentLabelmapImageToWorldMatrix = vtk.vtkMatrix4x4()
-        selectedSegmentLabelmap.GetImageToWorldMatrix(selectedSegmentLabelmapImageToWorldMatrix)
-        modifierImage.SetGeometryFromImageToWorldMatrix(selectedSegmentLabelmapImageToWorldMatrix)
-        # We could use a single slicer.vtkSlicerSegmentationsModuleLogic.ImportLabelmapToSegmentationNode
-        # method call to import all the resulting segments at once but that would put all the imported segments
-        # in a new layer. By using modifySegmentByLabelmap, the number of layers will not increase.
-        self.scriptedEffect.modifySegmentByLabelmap(segmentationNode, segmentID, modifierImage, slicer.qSlicerSegmentEditorAbstractEffect.ModificationModeSet)
+    islandCount = islandMath.GetNumberOfIslands()
+    islandOrigCount = islandMath.GetOriginalNumberOfIslands()
+    ignoredIslands = islandOrigCount - islandCount
+    logging.info( "%d islands created (%d ignored)" % (islandCount, ignoredIslands) )
+
+    baseSegmentName = "Label"
+    selectedSegmentID = self.scriptedEffect.parameterSetNode().GetSelectedSegmentID()
+    segmentationNode = self.scriptedEffect.parameterSetNode().GetSegmentationNode()
+    segmentation = segmentationNode.GetSegmentation()
+    selectedSegment = segmentation.GetSegment(selectedSegmentID)
+    selectedSegmentName = selectedSegment.GetName()
+    if selectedSegmentName is not None and selectedSegmentName != "":
+      baseSegmentName = selectedSegmentName
+
+    labelValues = vtk.vtkIntArray();
+    slicer.vtkSlicerSegmentationsModuleLogic.GetAllLabelValues(labelValues, islandImage);
+
+    # Erase segment from in original labelmap.
+    # Individuall islands will be added back later.
+    threshold = vtk.vtkImageThreshold()
```

Comment:
For future reference, it might be a bit cleaner to initialize using Fill() method instead of threshold filter:

```
imageData = vtk.vtkImageData()
imageData.SetDimensions(imageSize)
imageData.AllocateScalars(voxelType, 1)
imageData.GetPointData().GetScalars().Fill(fillVoxelValue)
```